applications/launchpad/gui-react/src/types/general.ts

```diff
@@ -5,6 +5,25 @@ export type CoinType = 'xtr' | ' xmr'
 
 export type MiningNodeType = 'tari' | 'merged'
 
+export type Time = {
+  hours: number
+  minutes: number
+}
+
+export type Interval = {
+  from: Time
+  to: Time
+}
+
+export type Schedule = {
+  id: string
+  enabled: boolean
+  days?: number[]
+  date?: Date
+  interval: Interval
+  type: MiningNodeType[]
+}
```

Comment:
Idea: _(sharing my thoughts that we can discuss later, not necessary any change request/suggestion)_

I'm wondering if there is anything else that may require scheduling. The only thing I've recognised is a periodical checking of new Docker images (and maybe availability of the new app version). However, it's something that can be also handled on the app launch, so it not necessary has to be scheduled.

Nevertheless, we may consider two improvements here:
1. having dedicated store/reducer for scheduling (instead having this in `app`)
2. adding field `scheduleType` or `eventType` that would define what action needs to be run. This way, we can schedule any event. (start mining, check images etc.).

Reply:
I haven't really thought about other schedules, in my mind periodical checks are a different beast - we would probably have a set period, configurable per build, so I don't think it's needed to have them stored as schedules in redux store